core/predicate_check.go

```diff
@@ -0,0 +1,44 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package core
+
+import (
+	"github.com/ava-labs/avalanchego/snow"
+	"github.com/ava-labs/avalanchego/snow/engine/snowman/block"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/params"
+	"github.com/ava-labs/subnet-evm/utils"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// CheckPredicatesForSenderTxs checks the stateful precompile predicates of any of the given
+// transactions that reference a stateful precompile address in their access list.
+// The parameter [txs] represents a flattened, nonce-ordered list of transactions originating from the same sender
+// Returns [len(txs), nil] if and only if all referenced predicates are met.
+// Otherwise, returns the index of the first transaction that was invalid and the predicate error.
+// In the failure case, all transactions after and including the index that failed the predicate should be considered invalid,
+// since the input [txs] are nonce-ordered and from the same sender.
```

Comment:
There is nothing in this function that requires the nonces to be in order from the same address, could we remove this comment?

It can be used this way because it will return the index of the first transaction that is invalid, but this is up to the caller, so I don't think the comment is necessary.

Reply:
I believe I suggested adding this comment previously. It wasn't clear to me why this function returned an index to the first failed transaction until I looked at PendingWithPredicates.

I agree that it is more appropriate for the comment to be at the caller. It looks like the requirement is already explained in the comments above PendingWithPredicates (I'm not sure if those comments were there when I made my original suggestion).